jobs/integration/test_opa.py

```diff
@@ -0,0 +1,408 @@
+import asyncio
+import json
+import random
+from pathlib import Path
+
+import pytest
+
+from .logger import log
+from .utils import JujuRunError, kubectl, scp_to, wait_for_application_status
+
+templates = Path(__file__).parent / "templates"
+
+
+class TestOPA:
+    @pytest.fixture(autouse=True)
+    def setup(self, model, k8s_model, tools):
+        self.model = model
+        self.k8s_model = k8s_model
+        self.tools = tools
+        control_plane_app = self.model.applications["kubernetes-control-plane"]
+        self.control_plane_unit = control_plane_app.units[0]
+
+    async def validate_create_ns_with_label(self):
+        remote_path = "/tmp/validate-gatekeeper-policy.yaml"
+        await scp_to(
+            templates / "validate-gatekeeper-policy.yaml",
+            self.control_plane_unit,
+            remote_path,
+            self.tools.controller_name,
+            self.tools.connection,
+        )
+
+        # Create the namespace
+        cmd = f"apply -f {remote_path}"
+        log("Creating namespace")
+        await kubectl(self.model, cmd)
```

Comment:
maybe there should be a util method called `kubectl_apply(document: Path)` which can do this for you.  Seems this is repeated quite a bit.

Reply:
and perhaps its analog is `kubectl_delete(document: Path)`